R/imageutils.R

```diff
@@ -50,9 +50,12 @@ plotPNG <- function(func, filename=tempfile(fileext='.png'),
   # guarantee user's code in func() will not trigger the error -- they may have
   # to set par(mar = smaller_value) before they draw base graphics.
   op <- graphics::par(mar = rep(0, 4))
-  tryCatch(graphics::plot.new(), finally = graphics::par(op))
+  on.exit(graphics::par(op), add = TRUE)
+  graphics::plot.new()
+
   dv <- grDevices::dev.cur()
-  tryCatch(shinyCallingHandlers(func()), finally = grDevices::dev.off(dv))
+  on.exit(grDevices::dev.off(dv), add = TRUE)
+  func()
```

Comment:
@yihui, do you know why doing `par(op)` on exit is too late? I just want to understand the root of the problem better. Does it close the device before running the `on.exit` stuff?

@jcheng5 reverting to the `tryCatch` code should be ok, right? That is, it shouldn't interfere with the debugging changes.